src/components/molecules/profile-view/profile-view.tsx

```diff
@@ -1,12 +1,22 @@
 import css from './profile-view.module.scss';
 import { Avatar } from '../../atoms/avatar/avatar';
 import { ProfileViewProps } from './profile-view.types';
+import { useNavigate } from '@tanstack/react-location';
 
 export const ProfileView = (props: ProfileViewProps): JSX.Element => {
-  const { type, img, name, size, location, theme = 'light' } = props;
+  const { type, img, name, username, size, location, theme = 'light' } = props;
+  const navigate = useNavigate();
+
+  function navigateToProfile() {
+    if (type === 'users') {
+      navigate({ to: `/profile/users/${username}/view` });
+    } else {
+      navigate({ to: `/profile/organizations/${username}/view` });
+    }
+  }
 
   return (
-    <div className={css.container}>
+    <div className={css.container} onClick={() => username && navigateToProfile()}>
```

Comment:
Please move the `onClick` logic to the avatar image as the `location` is of type `JSX.Element` and might be used as a link.

Having a link on the container when the child has a link will be problematic.